pandas/tests/test_index.py

```diff
@@ -2812,6 +2812,20 @@ def test_view(self):
         result = self._holder(i)
         tm.assert_index_equal(result, i)
 
+class DatetimeAbsoluteLike(DatetimeLike):
+
+    def test_contains(self):
+
+        i = self.create_index()
+
+        self.assertTrue(i[2] in i)
+        self.assertFalse('2012' in i)
+        #  python datetime objects work
+        self.assertTrue(datetime(2013,1,1) in i)
```

Comment:
add a test with a Timestamp and a Period as well (not sure how DatetimeIndex will handle the Period but try it)